eng/common/pipelines/templates/steps/credscan.yml

```diff
@@ -0,0 +1,41 @@
+parameters:
+  SuppressionFilePath: 'eng/CredScanSuppression.json'
+  BaselineFilePath: ''
+  SourceDirectory: $(Build.SourcesDirectory)
+
+steps:
+- pwsh: |
+    New-Item -Path "${{ parameters.SourceDirectory }}" -Name "credscan.tsv"
+    if ("$(Build.Reason)" -eq 'PullRequest') {
+      New-Item -Path "${{ parameters.SourceDirectory }}" -Name "changedFiles" -ItemType "directory"
+      (git diff "origin/$(System.PullRequest.TargetBranch)" HEAD --name-only)
+      | ForEach-Object { Add-Content -Path "${{ parameters.SourceDirectory }}/credscan.tsv" -Value "${{ parameters.SourceDirectory }}/$_"}
+    }
+    else {
+      Set-Content "credscan.tsv" -Value "${{ parameters.SourceDirectory }}/sdk/${{ parameters.ServiceDirectory }}"
+    }
+    Get-Content "${{ parameters.SourceDirectory }}/credscan.tsv"
+  displayName: Put changed files to folder     
+- task: securedevelopmentteam.vss-secure-development-tools.build-task-credscan.CredScan@2
+  displayName: Scanning for credentials
+  inputs:
+    toolMajorVersion: V2
+    toolVersion: latest
+    scanFolder: "${{ parameters.SourceDirectory }}/credscan.tsv"
+    suppressionsFile: ${{ parameters.SuppressionFilePath }}
+- task: securedevelopmentteam.vss-secure-development-tools.build-task-postanalysis.PostAnalysis@2
+  displayName: 'Post Analysis'
+  inputs:
+    GdnBreakBaselineFiles: ${{ parameters.BaselineFilePath }}
+    GdnBreakAllTools: false
+    GdnBreakGdnToolCredScan: true
+    GdnBreakGdnToolCredScanSeverity: Error
+    GdnBreakBaselines: baseline
+    # Used for generating baseline file.
+    # GdnBreakOutputBaselineFile: dotnet
+    # GdnBreakOutputBaseline: baseline
+  condition: succeededOrFailed()
+- pwsh: |
```

Comment:
Will people see this message? In cases where the post analysis steps fail how do we draw their attention to this step and message?

Reply:
Right, post analysis does not provide custom log message. If add the troubleshooting message like this, we need some education efforts.

Reply:
OK I guess this won't hurt but I suspect it might end up not being noticed.